Update skeleton.mdx (#3742)
grommet/hpe-design-system: aries-site/src/pages/components/skeleton.mdx

  <DashBoardSkeletonExample />
</Example>

### Use When
- Waiting for content to be returned and displayed to a user, to provide a sense of page structure and minimize layout shifts.

> #### Don't Use When

Reviewer: Is it worthwhile stating "Don't use for known, static content. Display them immediately while dynamic content is being retrieved."? 

I'm fine with leaving out to keep things simple. We only add it in in the future if needed (e.g. we see skeleton being applied to titles, headings, navigation, etc.). Thoughts?

```suggestion
#### Don't Use When
- Known, static content such as page titles, section headings, and navigational elements which are always shown and independent of data-driven content.
```

</Example>

### When to use

> - The interface is content-rich and has asynchronous loading, in order to provide the user with a sense of the page structure and to minimize layout shifts when content is loaded in.

Reviewer: Thinking about this last night... Would it be even simpler if we used the following as guidance?

"If waiting for content to be returned and displayed to a user, use skeleton to provide users with a sense of page structure and minimize layout shifts when content is available. 

If waiting for a process to complete, use busy button, progress bar, or spinner to communicate an action is taking place."

> ### When not to use

Reviewer: Since this is directly related to "When to use," should this be a child/subsection under "when to use" as opposed to a peer?

```suggestion
#### When not to use
```

> - The user is awaiting feedback for in-context actions such as fetching search results. Use a [Spinner](/components/spinner) instead.
> - The user is awaiting feedback on a submission process, like submitting a form. Use a [Busy Button](/components/button#busy-button) instead.

Reviewer: Modifying the search results use case. I'd argue that the structure of search results is known (i.e. we know the format in which the UI will present the search results) and should use skeleton. Spinners and progress bars should only be used when a process is being run by a service and there is no content being returned other than a success/error message.

I know these suggestions aren't as bullet-pointy. Feel free to play with the wording, but the criteria that should be used is:
- If content is being returned and how the UI presents that content is known, use skeleton.
- Use busy button, progress bar, or spinner to indicate a service or process is being executed and feedback is presented to users in the form of a success/error confirmation.

```suggestion
- Known, static content such as page titles, section headings, and navigational elements which are always shown and independent of data-driven content.
- The user is awaiting feedback on a submission process, like submitting a form. Use a [Busy Button](/components/button#busy-button) instead.
- The user is awaiting feedback for in-context actions such as initiating longer running processes. Instead, use a progress bar (coming soon) for determinate processes or [Spinner](/components/spinner) for indeterminate.
```

OR 

```suggestion
- Known, static content such as page titles, section headings, and navigational elements which are always shown and independent of data-driven content.
- When the purpose of the feedback is to communicate that a service or process is running and the user desires confirmation of successful completion. Instead, use a [Busy Button](/components/button#busy-button) for a submission process such as submitting a form; For longer running processes, use a progress bar (coming soon) for determinate processes or [Spinner](/components/spinner) for indeterminate.
```

OR

```suggestion
- Known, static content such as page titles, section headings, and navigational elements which are always shown and independent of data-driven content.
- A user desires feedback communicating a service or process is running and desires confirmation of successful completion. Use [busy button, progress bar, or spinner](link to the progress indicators page you are creating which will provide the guidance for when to use each) instead.
```

</Example>

### When to use

> - The interface is content-rich, with numerous elements such as images, text, and other media that take time to load and are known.

Reviewer 2: It feels like we could consolidate these 2 bullets into one concise one:

- When data has not yet loaded use Skeleton to provide a scaffold for the page to allow the user have a sense of the page structure and to minimize layout shifts when content is loaded in.

Author: I think using the "content-rich" expression might help the users understand a bit better the difference between skeleton and spinner. So, how about this: 
When to use:
- The interface is content-rich and has asynchronous loading, to stabilize layout and prevent shifts during content retrieval.

Reviewer 2: A happy medium?

- When the interface is content-rich and has asynchronous loading, use Skeleton to provide the user with a sense of the page structure and to minimize layout shifts when content is loaded in.

- The content is fetched asynchronously, especially from different sources or APIs. Using a skeleton can maintain the structure of the page and prevent layout shifts when content starts loading in multiple regions.

### When not to use

> - The user is awaiting feedback for in-context actions such as fetching search results. Use a [Spinner] (/components/spinner) instead.

Reviewer 2: ```suggestion
- The user is awaiting feedback for in-context actions such as fetching search results. Use a [Spinner](/components/spinner) instead.
```

> - The user is awaiting feedback on a submission process, like submitting a form. Use a [Busy Button] (/components/button?q=button#busy-button) instead.

Reviewer 2: ```suggestion
- The user is awaiting feedback on a submission process, like submitting a form. Use a [Busy Button](/components/button#busy-button) instead.
```
